### PR TITLE
9802 followup to remove excess CompName and CompType in SizeFanCoilUnit

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes an issue where the CompName and CompType of the FCU heating coil was passed to sizing. The function SizeFanCoilUnit is sizing the fan coil unit, not the heating coil.

 - This cleans up an issue found in #9802 where sizing the fan coil used the heating coil name.
 - 0 diffs should occur for merging.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/FanCoilUnits.cc [hunk 1]
```diff
@@ -1720,8 +1720,6 @@ namespace FanCoilUnits {
             }
         } else if (fanCoil.HCoilType_Num == HCoil::Electric) {
             if (fanCoil.DesignHeatingCapacity == DataSizing::AutoSize) {
-                CompName = fanCoil.HCoilName;
-                CompType = fanCoil.HCoilType;
```
Comment (rraustad) [inline src/EnergyPlus/FanCoilUnits.cc:1724]: Here the sizing routines are passed the heating coil name and type when the heating coil is not being sized.

File: src/EnergyPlus/FanCoilUnits.cc [hunk 2]
```diff
@@ -1901,8 +1899,6 @@ namespace FanCoilUnits {
 
             if (fanCoil.CapCtrlMeth_Num == CCM::ASHRAE && !fanCoil.ASHRAETempControl) {
 
-                CompType = fanCoil.UnitType;
-                CompName = fanCoil.Name;
```
Comment (rraustad) [inline src/EnergyPlus/FanCoilUnits.cc:1905]: duplicates

File: src/EnergyPlus/GeneralRoutines.cc [hunk 3]
```diff
@@ -666,7 +666,7 @@ void CheckThisAirSystemForSizing(EnergyPlusData &state, int const AirLoopNum, bo
 
 void CheckZoneSizing(EnergyPlusData &state,
                      std::string_view const CompType, // Component Type (e.g. Chiller:Electric)
-                     std::string const &CompName      // Component Name (e.g. Big Chiller)
+                     std::string_view const CompName  // Component Name (e.g. Big Chiller)
```
Comment (rraustad) [inline src/EnergyPlus/GeneralRoutines.cc:669]: No reason these can't be string_view when these args are only used in formatted warnings.

**Comment by rraustad** [comment]
The diff here must be because this branch that was current this morning is now behind develop. For that reason I'll fix the clang issue and pull develop later tonight.

**Comment by Myoldmopar** [comment]
> I'll fix the clang issue and pull develop later tonight.

Sounds good.  I am going to be doing with my merge binge in the next couple hours, so anytime late in workday hours should be a safe bet for pulling in develop and getting clean results.

**Comment by nrel-bot-3** [comment]
@rraustad @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 27 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 12 days since this pull request was last updated.

**Review by Myoldmopar — COMMENTED** [review]
This is a fine little cleanup @rraustad, always love to see more string views.  And it seems that you are no longer overwriting the CompType/CompName variables with HCoil info, which is the fix you are looking for.  Is this ready for review?

**Comment by rraustad** [comment]
@Myoldmopar this change caused diffs in Coil Sizing Details which would need to be investigated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 9 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 18 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 11 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 13 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 9 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 8 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
@rraustad could you comment on this PR?  I'm fine with closing it, leaving it as a draft, or pushing on it if it's worth pushing on.  If you don't see much value in keeping it around and actively getting pinged about it, feel free to close it.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 10 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 13 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 12 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 10 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 10 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 11 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@rraustad it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@rraustad it has been 15 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 10 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 9 days since this pull request was last updated.

**Comment by rraustad** [comment]
Closing as out of date

**Pull request closed without merging**